### PR TITLE
Backwards compatibility fixed

### PR DESCRIPTION
**Pull request by kj-crypto**

### CODE REVIEW

File: electrum/three_keys/utils.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def update_tx_status(tx_hash, tx_mined_info: 'TxMinedInfo', status: int, status_str: str, db) -> Tuple[int, str]:
+    if status_str == 'unknown':
+        return status, status_str
+
+    tx = db.get_transaction(tx_hash)
+
+    if tx.tx_type in TX_TYPES_LIKE_STANDARD:
+        if tx.tx_type == TxType.ALERT_CONFIRMED:
```
Comment (harzo) [inline electrum/three_keys/utils.py:21]: shouldn't `ALERT_RECOVERED` be handled?

File: electrum/three_keys/utils.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+from typing import Tuple
+
+from .tx_type import TxType
+
+
+TX_STATUS_INDEX_SHIFT = 10
+TX_TYPES_LIKE_STANDARD = (
+    TxType.NONVAULT,
+    TxType.INSTANT,
+    TxType.ALERT_CONFIRMED,
```
Comment (harzo) [inline electrum/three_keys/utils.py:10]: why `ALERT_PENDING` and `ALERT_RECOVERED` are missing?

File: electrum/address_synchronizer.py [hunk 3]
```diff
@@ -723,20 +730,39 @@ def get_addr_io(self, address):
...
+                    else:
+                        received[key] = value
+            for tx_hash, height, tx_type, *__ in h:
                 l = self.db.get_txi_addr(tx_hash, address)
                 for txi, v in l:
-                    sent[txi] = height
-        return received, sent
+                    if tx_type == TxType.ALERT_PENDING.name:
+                        alert_outgoing[txi] = height
+                    elif tx_type == TxType.ALERT_CONFIRMED.name:
```
Comment (harzo) [inline electrum/address_synchronizer.py:752]: Regarding incoming ElectrumX changes change this line to:
`elif tx_type == TxType.ALERT_CONFIRMED.name or tx_type == TxType.ALERT_RECOVERED.name:`

File: electrum/address_synchronizer.py [hunk 4]
```diff
@@ -723,20 +730,39 @@ def get_addr_io(self, address):
...
-            for tx_hash, height, *__ in h:
+            alert_incoming = {}
+            alert_outgoing = {}
+            for tx_hash, height, tx_type, *__ in h:
+                for n, v, is_cb in self.db.get_txo_addr(tx_hash, address):
+                    key = tx_hash + ':%d' % n
+                    value = (height, v, is_cb)
+                    if tx_type == TxType.ALERT_PENDING.name:
+                        alert_incoming[key] = value
+                    elif tx_type == TxType.ALERT_CONFIRMED.name:
```
Comment (harzo) [inline electrum/address_synchronizer.py:741]: Regarding incoming ElectrumX changes change this line to:
`elif tx_type == TxType.ALERT_CONFIRMED.name or tx_type == TxType.ALERT_RECOVERED.name:`

File: electrum/address_synchronizer.py [hunk 5]
```diff
@@ -338,35 +339,41 @@ def get_depending_transactions(self, tx_hash):
...
             if tx is None:
                 continue
+            self._mutate_transaction_type(current_tx=tx, incoming_type=TxType.from_str(tx_type))
             self.add_transaction(tx_hash, tx, allow_unrelated=True)
 
         # Store fees
         for tx_hash, fee_sat in tx_fees.items():
             self.db.add_tx_fee_from_server(tx_hash, fee_sat)
 
+    def _mutate_transaction_type(self, current_tx: 'ThreeKeysTransaction', incoming_type: TxType):
```
Comment (harzo) [inline electrum/address_synchronizer.py:372]: will it always be ThreeKeysTransaction? remember that from now on this logic happens in all wallet types

**Comment by harzo** [comment]
Also this line
https://github.com/bitcoinvault/electrum-vault/blob/bugfix/atx-in-standard-wallet/electrum/verifier.py#L94
 should be changed to:
`if tx_type == TxType.ALERT_PENDING.name or tx_type == TxType.ALERT_RECOVERED.name:`

File: electrum/three_keys/utils.py (hunk 2 above)
Comment (kj-crypto) [reply electrum/three_keys/utils.py:10]: this tuple contains types which will be displayed like NonValut transaction, so `ALERT_PENDING` and `ALERT_RECOVERED` should be missed

File: electrum/address_synchronizer.py (hunk 5 above)
Comment (kj-crypto) [reply electrum/address_synchronizer.py:372]: every fetched transaction via `self.network.get_transaction(tx_hash)` is being casted into `ThreeKeysTransaction`

File: electrum/three_keys/utils.py (hunk 1 above)
Comment (kj-crypto) [reply electrum/three_keys/utils.py:21]: `ALERT_RECOVERED` is handled in return statement of this method, like other transaction types

**Pull request merged**